frontend/pages/layout.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { usePathname } from 'next/navigation';
+import Header from '@/components/header';
+import Footer from '@/components/footer';
+
+const showHeaderFooter = (pathname: string) => !['/login', '/register'].includes(pathname);
+
+export default function Layout({ children }: { children: React.ReactNode }) {

```

Comment:
`PropsWithChildren`이라는 리액트에서 사전정의한 타입이 있습니다. 그 타입을 이용하면 children을 사용 할 수 있습니다. 

```suggestion
import Header from '@/components/header';
import Footer from '@/components/footer';

const showHeaderFooter = (pathname: string) => !['/login', '/register'].includes(pathname);

export default function Layout({ children }: PropsWithChildren) {

```